api/rest/client/methods.go

```diff
@@ -172,3 +177,37 @@ func (c *Client) GetConnectGraph() (api.ConnectGraphSerial, error) {
 	err := c.do("GET", "/health/graph", nil, &graphS)
 	return graphS, err
 }
+
+// WaitFor is a utility function that allows for a caller to
+// wait for a paticular status for a CID. It returns a channel
+// upon which the caller can wait for the targetStatus.
+func (c *Client) WaitFor(ci *cid.Cid, local bool, target api.TrackerStatus, checkFreq time.Duration) <-chan api.TrackerStatus {
+	statusCh := make(chan api.TrackerStatus)
+
+	go func(statusCh chan api.TrackerStatus) {
+		for {
+			gblPinInfo, err := c.Status(ci, local)
+			if err != nil {
+				// TODO(ajl): check how logging is done in ipfs-cluster
+				logger.Errorf("failed to get pin status: %v", err)
+				// TODO(ajl): check I can override api.TrackerStatusBug here for
+				// error reporting purposes or return an error chan,
+				// not sure which is the best option.
+				statusCh <- api.TrackerStatusBug
+			}
+			for _, pinInfo := range gblPinInfo.PeerMap {
+				if target != pinInfo.Status {
```

Comment:
You have to account for `Remote` status. Essentially, what you want to check for pinned is that "all pins have status PINNED except those with REMOTE or ERROR.". Perhaps worth returning an error if you see a pin in error state and aborting the whole thing. If no one is in ERROR, it amounts to checking that things are PINNED (excluding REMOTE), or UNPINNED.